### PR TITLE
Theme previews

### PR DESCRIPTION
**Pull request by canstudios-louisem**
This will search for a file called preview.jpg in the root of your theme and display it on the theme selection screen if there is one.

I will also make a PR to the vanilla theme with a pr for the preview.jpg one of our designers made.

### CODE REVIEW

File: frontend/src/modules/editor/selectTheme/less/editorTheme.less [hunk 1]
```diff
@@ -56,7 +62,7 @@
     width: 60px;
     float: left;
     position: relative;
-    background: url(/css/assets/theme_bg.png) top left no-repeat;
+    background: url(/images/theme_bg.png) top left no-repeat;
```
Comment (taylortom) [inline frontend/src/modules/editor/selectTheme/less/editorTheme.less]: Is this correct?

File: frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js [hunk 2]
```diff
@@ -24,6 +24,17 @@ define(function(require){
...
 
     postRender: function() {
+      var previewUrl = '/api/theme/preview/' + this.model.attributes.name + '/' + this.model.attributes.version;
+      var $previewLoc = this.$el.find('.theme-preview');
+
+      $.ajax(previewUrl, {
+        statusCode: {
+          // if a preview image is found
+          200: function() {
+            $previewLoc.prepend('<img src="' + previewUrl + '"alt="Preview image"' + '/>');
```
Comment (tomgreenfield) [inline frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js]: Can the alt text be localised please?

Could also switch to the jQuery object syntax if you fancied it:
```js
$previewLoc.prepend($('<img/>', { src: previewUrl, alt: '...' }));
```

**Comment by taylortom** [comment]
On accessing the theme picker page, I get the following console errors:
```
11:17:58.042 jquery.js:9659 GET http://localhost:5000/api/theme/preview/adapt-theme-community/0.0.2 404 (Not Found)
11:17:58.047 jquery.js:9659 GET http://localhost:5000/api/theme/preview/adapt-contrib-vanilla/2.0.4 404 (Not Found)
```

**Comment by taylortom** [comment]
Also a slightly cheeky request, but could you also fix the theme's description text colour for the selected state? I'm sure I've done this before, but may have been lost in a merge. We just need it to go white like the title really.

**Comment by canstudios-louisem** [comment]
I will change the 404 response to a 204 would be fine and that won't give a console error and is a standard reponse for this kind of thing.

File: frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js [hunk 3]
```diff
@@ -24,6 +24,16 @@ define(function(require){
     },
 
     postRender: function() {
+      var previewUrl = '/api/theme/preview/' + this.model.attributes.name + '/' + this.model.attributes.version;
```
Comment (tomgreenfield) [inline frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js]: Very minor but could we use `model.get()` to access attributes here please, for consistency.

**Comment by canstudios-louisem** [comment]
@taylortom @tomgreenfield I have made those changes.

**Comment by canstudios-louisem** [comment]
Before release we also need to update https://github.com/adaptlearning/adapt_framework/wiki/Developers-Guide:-Theme

**Comment by canstudios-louisem** [comment]
@tomgreenfield I've just tested 16:9 and it's quite short which means you can't really see any components and the header I think we should stay at 275x205. I will add the same default icon as in assets though.

**Comment by canstudios-louisem** [comment]
@tomgreenfield and @taylortom do you want to re approve as I've pushed a few things since you did?

File: frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js [hunk 4]
```diff
@@ -24,6 +24,19 @@ define(function(require){
...
+      var previewUrl = '/api/theme/preview/' + this.model.get('name') + '/' + this.model.get('name')
+      var $previewLoc = this.$('.theme-preview');
+
+      $.ajax(previewUrl, {
+        statusCode: {
+          200: function() {
+            $previewLoc.prepend($('<img/>', { src: previewUrl, alt: Origin.l10n.t('app.themepreviewalt') }));
+          },
+          204: function() {
+            $previewLoc.prepend($('<i/>', { class: 'fa fa-file-image-o' }));
```
Comment (taylortom) [inline frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js]: Very minor suggestion: I wonder if we can use something more theme-related for the icon here, as we use the image file one in various other places in the app's asset management. Best font awesome icon I can find is the paintbrush (there may be something better):

screen shot 2017-08-21 at 11 39 39

Comment (lc-thomasberger) [inline frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js]: image icon looks more like it is related to assets

Comment (canstudios-louisem) [inline frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js]: That's fine by me ive asked one of our creatives to see if they know anything better than the paintbrush as im not sure its quite right.

Comment (taylortom) [inline frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js]: Cool, thanks Louise!

Comment (tomgreenfield) [inline frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js]: We need quotes around `class` since it's a reserved word in JavaScript.

**Comment by canstudios-louisem** [comment]
@taylortom @tomgreenfield I've pushed an update with those changes. I added the paintbrush but our designer also made [this](http://i.imgur.com/uMvlKD3.png) thoughts? I'm not set on either the paintbrush is one less image file tho.

**Comment by taylortom** [comment]
Great thanks @canstudios-louisem, looks good. 

I have to admit that the linked image icon doesn't mean anything to me, I'd never guess it was related to appearance/theming. Happy to go with the general consensus though.

File: frontend/src/modules/editor/selectTheme/less/editorTheme.less [hunk 5]
```diff
@@ -20,33 +20,51 @@
   transition:all 0.3s;
   border: 1px solid #d6f2f9;
 
+  &-inner {
+    .theme-header {
+      padding: 20px 0;
```
Comment (tomgreenfield) [inline frontend/src/modules/editor/selectTheme/less/editorTheme.less]: Can the padding be made equal so we avoid text hitting the side of the container? We then won't need it declared for `.display-name` or `.description`.

padding

File: frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js [hunk 6]
```diff
@@ -24,6 +24,19 @@ define(function(require){
...
+      var previewUrl = '/api/theme/preview/' + this.model.get('name') + '/' + this.model.get('name');
+      var $previewLoc = this.$('.theme-preview');
+
+      $.ajax(previewUrl, {
+        statusCode: {
+          200: function() {
+            $previewLoc.prepend($('<img/>', { src: previewUrl, alt: Origin.l10n.t('app.themepreviewalt') }));
+          },
+          204: function() {
+            $previewLoc.prepend($('<i/>', { 'class': 'fa fa-paint-brush' }));
```
Comment (tomgreenfield) [inline frontend/src/modules/editor/selectTheme/views/editorThemeItemView.js]: Happy with the paintbrush icon, @taylortom.

**Comment by canstudios-louisem** [comment]
sorted @tomgreenfield

**Pull request merged**